Modules/xSharePoint/DSCResources/MSFT_xSPWebAppPolicy/MSFT_xSPWebAppPolicy.psm1

```diff
@@ -125,16 +347,108 @@ function Test-TargetResource
     param
     (
         [parameter(Mandatory = $true)]  [System.String] $WebAppUrl,
-        [parameter(Mandatory = $true)]  [System.String] $UserName,
-        [parameter(Mandatory = $true)]  [ValidateSet("Deny All","Deny Write","Full Read", "Full Control")] [System.String] $PermissionLevel,
-        [parameter(Mandatory = $false)] [System.Boolean] $ActAsSystemUser,
+        [parameter(Mandatory = $false)] [Microsoft.Management.Infrastructure.CimInstance[]] $Members,
+        [parameter(Mandatory = $false)] [Microsoft.Management.Infrastructure.CimInstance[]] $MembersToInclude,
+        [parameter(Mandatory = $false)] [Microsoft.Management.Infrastructure.CimInstance[]] $MembersToExclude,
+        [parameter(Mandatory = $false)] [System.Boolean] $SetCacheAccountsPolicy,
         [parameter(Mandatory = $false)] [System.Management.Automation.PSCredential] $InstallAccount
     )
 
     $CurrentValues = Get-TargetResource @PSBoundParameters
     Write-Verbose -Message "Testing web app policy for $UserName at $WebAppUrl"
     if ($null -eq $CurrentValues) { return $false }
-    return Test-xSharePointSpecificParameters -CurrentValues $CurrentValues -DesiredValues $PSBoundParameters -ValuesToCheck @("PermissionLevel", "ActAsSystemUser")
+
+    if (($SetCacheAccountsPolicy) -and (-not($CurrentValues.ContainsKey("SuperUserAccount")) -or -not($CurrentValues.ContainsKey("SuperReaderAccount")))) {
```

Comment:
Having a get method that doesn't match the schema will cause some errors in PowerShell though - so if you need them in the test method you might need to retrieve them again there instead.